src/main/java/io/cronitor/client/CronitorPinger.java

```diff
@@ -0,0 +1,81 @@
+package io.cronitor.client;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.logging.Logger;
+
+import io.cronitor.client.urlgenerators.CommandUrlGenerator;
+
+public class CronitorPinger {
+
+    public CronitorPinger() {
+
+        this.connection = null;
+    }
+
+
+
+    HttpURLConnection connection;
+    URL url;
```

Comment:
Even if it's private by default, it's good to write it down anyway (for future reviewers understanding or Java version we never know)

Reply:
yeah i hear ya, these are currently exposed to the package for testing purposes. (see CronitorPingerTest.java). Not knowing the tooling well I really struggled with these tests. And have not been able to mock a URL object in order to simulate timeouts and error codes. Any tips/helps with this would be great!